accounts-db/src/read_only_accounts_cache.rs

```diff
@@ -202,13 +230,23 @@ impl ReadOnlyAccountsCache {
     }
 
     pub(crate) fn remove(&self, pubkey: Pubkey, slot: Slot) -> Option<AccountSharedData> {
-        let (_, entry) = self.cache.remove(&(pubkey, slot))?;
+        Self::do_remove(&(pubkey, slot), &self.cache, &self.queue, &self.data_size)
+    }
+
+    /// Removes `key` from the cache, if present, and returns the removed account
+    fn do_remove(
+        key: &ReadOnlyCacheKey,
+        cache: &DashMap<ReadOnlyCacheKey, ReadOnlyAccountCacheEntry>,
+        queue: &Mutex<IndexList<ReadOnlyCacheKey>>,
+        data_size: &AtomicUsize,
+    ) -> Option<AccountSharedData> {
+        let (_, entry) = cache.remove(key)?;
         // self.queue should be modified only after removing the entry from the
         // cache, so that this is still safe if another thread writes to the
         // same key.
-        self.queue.lock().unwrap().remove(entry.index());
-        let account_size = self.account_size(&entry.account);
-        self.data_size.fetch_sub(account_size, Ordering::Relaxed);
+        queue.lock().unwrap().remove(entry.index());
+        let account_size = Self::account_size(&entry.account);
+        data_size.fetch_sub(account_size, Ordering::Relaxed);
```

Comment:
should we change the ordering to "Released" since we are no longer running on the same thread?

Reply:
And "Required" for load?